packages/roosterjs-content-model/lib/modelApi/block/setModelAlignment.ts

```diff
@@ -0,0 +1,80 @@
+import { ContentModelDocument } from '../../publicTypes/group/ContentModelDocument';
+import { ContentModelListItem } from '../../publicTypes/group/ContentModelListItem';
+import { ContentModelTable } from '../../publicTypes/block/ContentModelTable';
+import { getOperationalBlocks } from '../selection/collectSelections';
+import { isBlockGroupOfType } from '../common/isBlockGroupOfType';
+
+const ResultMap: Record<
+    'left' | 'center' | 'right',
+    Record<'ltr' | 'rtl', 'start' | 'center' | 'end'>
+> = {
+    left: {
+        ltr: 'start',
+        rtl: 'end',
+    },
+    center: {
+        ltr: 'center',
+        rtl: 'center',
+    },
+    right: {
+        ltr: 'end',
+        rtl: 'start',
+    },
+};
+
+/**
+ * @internal
+ */
+export function setModelAlignment(
+    model: ContentModelDocument,
+    alignment: 'left' | 'center' | 'right'
+) {
+    const paragraphOrListItemOrTable = getOperationalBlocks<ContentModelListItem>(
+        model,
+        ['ListItem'],
+        ['TableCell']
+    );
+
+    paragraphOrListItemOrTable.forEach(({ block }) => {
+        if (isBlockGroupOfType<ContentModelListItem>(block, 'ListItem')) {
+            block.formatHolder.format.alignSelf =
+                ResultMap[alignment][block.format.direction == 'rtl' ? 'rtl' : 'ltr'];
+            block.levels.forEach(level => {
+                level.flexDirection = 'column';
+                level.display = 'flex';
+            });
+        } else if (block.blockType === 'Table') {
+            const isWholeTableSelected = block?.cells.every(row =>
```

Comment:
I think when table is returned, it must be full table selected, so no need to check here.

You can verify.